torchvision/prototype/features/_bounding_box.py

```diff
@@ -120,20 +118,14 @@ def pad(
         fill: Optional[Union[int, float, Sequence[int], Sequence[float]]] = None,
         padding_mode: str = "constant",
     ) -> BoundingBox:
-        if padding_mode not in ["constant"]:
-            raise ValueError(f"Padding mode '{padding_mode}' is not supported with bounding boxes")
-
```

Comment:
Padding an image with other non-const modes will create new objects, but we won't be able to create corresponding bboxes, that's why we raise an error. Ignoring this case may lead to silent issues, I think

Reply:
You are right to say that if you use reflect, you won't be able to create bboxes for the new edges (maybe we should find a way to do it on the future). This might cause issues or not, we can't tell for sure as in Object Detection datasets there are often many unlabelled objects in the scene already.

The reason I believe this needs to be fixed is because the methods on the _Feature objects are just thin wrappers and should add no new functionality. There are there to redirect to the `F` kernel, rewrap the result and update the meta-data. Adding new functionality means diverging from the behaviour of `F` equivalent kernels. This is why we agreed that the method should receive maximally all parameters and use those that make sense without throwing errors. You can see this in action on how SegmentationMask handles wrong interpretation values. Here we should do the same here.

If we want to detect this and throw an error we need to put the check on the Pad transform. There we can ensure that specific modes aren't used when bboxes are present.

Reply:
As discussed offline with @vfdev-5, this check should move to the Kernel. I have sent a PR that does this but feel free to split across subPRs.

For reference what we decided is to validate input always on the kernel side. We also propose to ignore params only if they don't make sense for a given type (for example interpolation for SegmentationMask). Params which are unsupported should throw the right message on the lowlevel kernel side